File: Question3/IsSubsequence.java
package Question3;

public class IsSubsequence {
    public static boolean isSubsequence(String s, String t) {
        int i = 0, j = 0;
        while (i < s.length() && j < t.length()){
            char ch = s.charAt(i);
            char ch1 = t.charAt(j);
            if(ch == ch1){
                i++;
                j++;
            }else{
                j++;
            }
        }
        if (i == s.length()) return true;
        return false;
    }
    public static void main(String[] args) {
        String s = "abc";
        String t = "ahbgdc";
        System.out.println(isSubsequence(s, t));
    }
}

File: Question3/IsSubsequence.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i{0},j{0};
        while (i < s.size() && j < t.size()){
            // if(s[i] == t[j]){
            //     i += 1;
            //     j += 1;
            // }else {
            //     j += 1;
            // }
            if (s[i] == t[j]) i += 1;
            j += 1;
        }

        if (i == s.length()) return true;
        return false;
    }
};

int main()
{
    string s = "abc";
    string t = "ahbgdc";
    Solution obj;
    cout << obj.isSubsequence(s,t);
    return 0;
}